airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -61,25 +61,25 @@ class SimpleRetriever(Retriever, HttpStream):
     primary_key: Optional[Union[str, List[str], List[List[str]]]]
     _primary_key: str = field(init=False, repr=False, default="")
     paginator: Optional[Paginator] = None
-    stream_slicer: Optional[StreamSlicer] = SinglePartitionRouter(parameters={})
+    stream_slicer: StreamSlicer = SinglePartitionRouter(parameters={})
     cursor: Optional[Cursor] = None
     disable_retries: bool = False
     message_repository: MessageRepository = NoopMessageRepository()
 
-    def __post_init__(self, parameters: Mapping[str, Any]):
-        self.paginator = self.paginator or NoPagination(parameters=parameters)
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        self._paginator = self.paginator or NoPagination(parameters=parameters)
+        self._last_response: Optional[requests.Response] = None
+        self._records_from_last_response: List[Record] = []
         HttpStream.__init__(self, self.requester.get_authenticator())
-        self._last_response = None
-        self._records_from_last_response = None
         self._parameters = parameters
-        self.name = InterpolatedString(self._name, parameters=parameters)
+        self._name = InterpolatedString(self._name, parameters=parameters) if isinstance(self._name, str) else self._name
```

Comment:
why not keep it as an interpolated string? it would avoid the need for the [if elses ](https://github.com/airbytehq/airbyte/pull/28386/files#diff-574e7f982441af95b199fa0d628677b9a654543eaea1534b7516c8679375a235R82)

Reply:
This line makes sure that `_name` is always an interpolated string